pennylane/qchem/factorization.py

```diff
@@ -278,8 +278,9 @@ def factorize(
 
         if cholesky and init_params is None:
             # compute the factors via cholesky decomposition routine
+            # For compressed factorization, we want exactly num_factors, not early stopping
             factors, f_eigvals, f_eigvecs = _double_factorization_cholesky(
-                two, tol_factor, shape, interface, num_factors
+                two, tol_factor=0.0, shape=shape, interface=interface, num_factors=num_factors
```

Comment:
Why are we changing the `tol_factor` here? 🤔

Reply:
somehow https://github.com/PennyLaneAI/pennylane/actions/runs/19584854941/job/56091452961 they will fail without a stricter tol